src/EditorFeatures/CSharpTest/CodeActions/Preview/PreviewTests.cs

```diff
@@ -102,7 +103,7 @@ public void TestPickTheRightPreview_NoPreference()
                 GetMainDocumentAndPreviews(workspace, out document, out previews);
 
                 // The changed document comes first.
-                var preview = previews.TakeNextPreview();
+                var preview = previews.TakeNextPreviewAsync().PumpingWaitResult();
```

Comment:
I had searched online and iirc come to the same conclusion (i.e. that async tests should work in Roslyn's version of xunit). However, when I tried searching within Roslyn.sln for examples of such tests in features / editor layer I couldn't find any. The tests that needed this functionality seemed to be using the .PumpingWait() or .PumpingWaitResult() helpers... I didn't spend too much time investigating since the above approach seems to work fine for these tests.